packages/back-end/test/services/settings/settings.test.ts

```diff
@@ -0,0 +1,251 @@
+          expect(metricSettings_testvar.regressionAdjustmentDays.value).toEqual(
+            12
+          );
+          expect(metricSettings_testvar.winRisk.value).toEqual(0.0015);
+          expect(metricSettings_testvar.loseRisk.value).toEqual(0.0225);
+
+          // CUPED denominator is count rejection
+          // const conversions_as_count: MetricInterface = {
+          //   ...metrics.conversions,
+          //   type: "count",
+          // };
+          // todo: how do we pass in this dependency while checking the `testvar` metric?
+          // Answer for Bryce: see `denominatorMetric` below
+          const { settings: metricSettings_testvar_2 } = useScopedSettings(
+            org.settings,
+            {
+              metric: metrics.testvar,
+              denominatorMetric: {
+                ...metrics.conversions,
+                type: "count",
+              },
+              experiment: experiments.exp1,
+            }
+          );
+          expect(
+            metricSettings_testvar_2.regressionAdjustmentEnabled.value
+          ).toEqual(false);
+          expect(
+            metricSettings_testvar_2.regressionAdjustmentEnabled.meta.reason
+          ).toEqual("denominator is count");
+        });
+      });
+    });
+
+    it("overrides stats-related metrics based on stats engine", () => {
+      const orgSettings1: Partial<OrganizationSettings> = {
+        statsEngine: "bayesian",
+        confidenceLevel: 0.95,
+        pValueThreshold: 0.05,
+      };
+      const org1 = genOrgWithSettings(orgSettings1);
+      const { settings: settings_revenue_1 } = useScopedSettings(
+        org1.settings,
+        {
+          metric: metrics.revenue,
+          experiment: experiments.exp1,
+        }
+      );
+
+      // org level:
+      expect(settings_revenue_1.regressionAdjustmentEnabled.value).toEqual(
+        false
+      );
+      expect(
+        settings_revenue_1.regressionAdjustmentEnabled.meta.reason
+      ).toEqual("stats engine is bayesian");
+      // TODO
+      // expect(settings_revenue_1.pValueThreshold.meta.warning).toEqual(
+      //   "stats engine is bayesian"
+      // );
+      expect(settings_revenue_1.confidenceLevel.value).toEqual(0.95);
+      // metric level:
+      expect(settings_revenue_1.winRisk.value).toEqual(0.0025);
+      expect(settings_revenue_1.loseRisk.value).toEqual(0.0125);
+
+      const orgSettings2: Partial<OrganizationSettings> = {
+        statsEngine: "frequentist",
+      };
+      const org2 = genOrgWithSettings(orgSettings2);
+      const { settings: settings_revenue_2 } = useScopedSettings(
+        org2.settings,
+        {
+          metric: metrics.revenue,
+          experiment: experiments.exp1,
+        }
+      );
+
+      // org level:
+      // Q for Bryce: exp1 does not have regressionAdjustmentEnabled defined for the metric override (check `met_r1`).
+      // does that need to be updated or should this change to expect false?
```

Comment:
Sorry I typoed that specific override. It's supposed to have:
```
    {
      id: "met_r1",
      regressionAdjustmentOverride: true,
      regressionAdjustmentEnabled: true,
      regressionAdjustmentDays: 8,
    },
```

Reply:
Updated